### PR TITLE
feat: introduce unique and alias identity record types

### PR DESCRIPTION
**Pull request by shumkov**

### CODE REVIEW

File: schema/dpns-contract-documents.json [hunk 1]
```diff
@@ -108,10 +112,10 @@
     "properties": {
       "saltedDomainHash": {
         "type": "string",
-        "minLength": 68,
-        "maxLength": 68,
-        "pattern": "^[A-Fa-f0-9]+$",
-        "description": "Double sha-256 multihash of the full domain name + salt in a form of a hex string"
+        "contentMediaType": "application/octet-stream",
+        "minLength": 34,
```
Comment (thephez) [inline schema/dpns-contract-documents.json]: What are the 2 extra bytes?

Comment (shumkov) [reply schema/dpns-contract-documents.json]: My bad. It supposed to be 32.

File: schema/dpns-contract-documents.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "minLength": 3,
+        "maxLength": 63,
+        "description": "Domain label. e.g. 'Bob'."
+      },
+      "normalizedLabel": {
+        "type": "string",
+        "pattern": "^((?!-)[a-z0-9-]{0,62}[a-z0-9])$",
+        "maxLength": 63,
+        "description": "Domain label in a lower case for case insensitive uniqueness validation. e.g. 'bob'",
+        "$comment": "This property will be deprecated due to case insensitive indices"
```
Comment (Alex-Werner) [inline schema/dpns-contract-documents.json]: What's makes us not deprecating it right now ? @shumkov

Comment (shumkov) [reply schema/dpns-contract-documents.json]: We don't have such indices atm :)

**Review by Alex-Werner — APPROVED** [review]
Nice ! Contract being documented and commented is really a good standard we set :D

File: schema/dpns-contract-documents.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
```
Comment (shumkov) [inline schema/dpns-contract-documents.json]: must be equal to lower label?

File: schema/dpns-contract-documents.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      },
+      "normalizedParentDomainName": {
+        "type": "string",
+        "pattern": "^$|^((?!-)[a-z0-9-\\.]{0,189}[a-z0-9])$",
+        "minLength": 0,
+        "maxLength": 190,
+        "description": "A full parent domain name in lower case for case insensitive uniqueness validation. e.g. 'dash'",
+        "$comment": "Must be equal to existing domain or can be empty if you want to create top level domain. Only top level Identity can create top level domains."
```
Comment (shumkov) [inline schema/dpns-contract-documents.json]: Only the contract owner ...

File: schema/dpns-contract-documents.json [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      },
+      "preorderSalt": {
+        "type": "string",
+        "contentEncoding": "binary",
+        "minLength": 32,
+        "maxLength": 32,
+        "description": "Domain pre-order salt. 32 random bytes",
+        "$comment": "Double sha-256 of concatenated salt is used for pre-ordering document."
```
Comment (shumkov) [inline schema/dpns-contract-documents.json]: this is just salt

File: schema/dpns-contract-documents.json [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      "records": {
+        "type": "object",
+        "properties": {
+          "dashUniqueIdentityId": {
+            "type": "string",
+            "minLength": 42,
+            "maxLength": 44,
+            "pattern": "^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]+$",
+            "description": "Identity ID string should be used to create a primary name for Identity",
+            "$comment": "base58 string"
```
Comment (shumkov) [inline schema/dpns-contract-documents.json]: should be equal to document owner

File: schema/dpns-contract-documents.json [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    "required": [
+      "label",
+      "normalizedLabel",
+      "normalizedParentDomainName",
+      "preorderSalt",
+      "records",
+      "subdomainRules"
+    ],
+    "additionalProperties": false,
+    "$comment": "normalizedLabel + '.' + normalizedParentDomain must be < 254 chars length"
```
Comment (shumkov) [inline schema/dpns-contract-documents.json]: I would explain why 254

File: schema/dpns-contract-documents.json [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    "properties": {
+      "saltedDomainHash": {
+        "type": "string",
+        "contentEncoding": "binary",
+        "minLength": 32,
+        "maxLength": 32,
+        "description": "Double sha-256 of concatenated salt, `normalizedLabel` and `normalizedParentDomainName`. Salt should be 32 random bytes"
+      }
+    },
+    "required": [
```
Comment (shumkov) [inline schema/dpns-contract-documents.json:124]: I would say that this doc is temporary one and can be removed after domain creation.

File: schema/dpns-contract-documents.json [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+      "saltedDomainHash"
+    ],
+    "additionalProperties": false
+  }
```
Comment (shumkov) [inline schema/dpns-contract-documents.json:130]: Let's put here the paragrah from dip that describes that you need to create a preorder and then domain and why

**Pull request merged**